model-optimizer/extensions/middle/StridedSliceNormalizer.py

```diff
@@ -0,0 +1,238 @@
+
+            PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:1', 'slice', 'dim_size')
+            PermuteInputs().set_input_permutation(node.in_node(2), node, 'input:2', 'slice', 'dim_size')
+            PermuteInputs().set_input_permutation(node.in_node(3), node, 'input:3', 'slice', 'dim_size')
+
+    @staticmethod
+    def normalize_strided_slice(graph: Graph, node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        input_rank = len(input_shape)
+        slice_rank = node.in_port(1).data.get_shape()[0]
+
+        StridedSlice.allign_mask_with_slice_rank(node, slice_rank)  # if StridedSlice is created after partial_infer
+        StridedSliceNormalizer.normalize_slices_attr(node)
+
+        num_insertations = input_rank - slice_rank + np.count_nonzero(node.new_axis_mask)
+        if np.any(node.ellipsis_mask):
+            assert len(np.nonzero(node.ellipsis_mask)) == 1, 'only one ellipsis_mask nonzero value is allowed'
+            ellipsis_start = np.nonzero(node.ellipsis_mask)[0][0]
+            # since we don't expect values in begin and end: take the whole range along ellipsis_start
+            node.begin_mask[ellipsis_start] = 0
+            node.end_mask[ellipsis_start] = 0
+            node.ellipsis_mask[ellipsis_start] = 0
+            insertation_start_idx = ellipsis_start + 1
+
+            StridedSliceNormalizer.unroll_ellipsis_for_inputs(graph, node, ellipsis_start, num_insertations)
+        elif num_insertations > 0:
+            insertation_start_idx = slice_rank  # insert blank values to mask ends
+            StridedSliceNormalizer.extend_inputs(node, num_insertations)
+
+        if num_insertations > 0:
+            # insert blank values for ellipsis unrolling and extending
+            for mask_name in StridedSlice.get_all_mask_names():
+                node[mask_name] = np.insert(node[mask_name], insertation_start_idx, [0] * num_insertations).astype(int)
+
+    @staticmethod
+    def unroll_ellipsis_for_inputs(graph: Graph, node: Node, ellipsis_start: int, num_ellipsis_ext: int):
+        node_name = node.soft_get('name', node.id)
+        for i, slice_name in enumerate(('begin', 'end', 'strides')):
+            i += 1
+            placeholder_arr = np.zeros(num_ellipsis_ext) if i != 3 else np.ones(num_ellipsis_ext)
+            placeholder_node = Const(graph, {'name': node_name + '/const_to_unroll_{}_ellipsis'.format(slice_name),
+                                             'value': int64_array(placeholder_arr)}).create_node()
+
+            concat_in_ports_count = 3 if ellipsis_start != 0 else 2
+            concat = Concat(graph, {'axis': 0, 'name': node_name + '/concat_{}'.format(slice_name),
+                                    'in_ports_count': concat_in_ports_count}).create_node()
+
+            if ellipsis_start != 0:
+                split = create_op_with_const_inputs(graph, VariadicSplit, {1: 0, 2: int64_array([ellipsis_start, -1])},
```

Comment:
`1: 0` should be changed to `1: int64_array(0)`

Reply:
done